### PR TITLE
implement DefaultV1 Recipe

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- fix https://github.com/RasaHQ/rasa/issues/9277
- implement recipe which converts current config to graph schemas
- logic to convert regular configs to train and predict graph schemas
- interpolates passed CLI params in to the graph components config
- handles Core only / NLU only training
- implemented decorator which allows to register classes with the `DefaultV1Recipe`

**Open Todos**
- [ ] wait until all graph components have been adapted to `GraphComponent` interface and then register all these classes with the decorator
- [ ] change the current `isinstance` checks to check the component type via the registered `ComponentType`

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by wochinge** [comment]
@joejuzl that's what I have so far

**Comment by wochinge** [comment]
The new edge case about "rule_only_data" would still need to be considered (see [here](https://rasa-hq.slack.com/archives/C01HMTEGX37/p1629877242006400))

**Comment by joejuzl** [comment]
@wochinge reminder: `JiebaTokenizer` persists and loads. (Although we may be able to avoid it.)

**Comment by wochinge** [comment]
reminder to pass language to Duckings model config if ducklings model config is not filled

**Comment by wochinge** [comment]
`EntitySynonymMapper` doesn't add anything during training which is required by subsequent components

**Comment by wochinge** [comment]
we also need autoconfig in there

File: rasa/core/policies/_ensemble.py [hunk 1]
```diff
@@ -74,6 +72,8 @@ def __init__(
         self._set_rule_only_data()
 
     def _set_rule_only_data(self) -> None:
+        from rasa.core.policies.rule_policy import RulePolicy
```
Comment (wochinge) [inline rasa/core/policies/_ensemble.py:75]: Changes here were needed to avoid cyclic imports

**Pull request closed without merging**